### PR TITLE
[refactor] Config, module base, and cleanup

### PR DESCRIPTION
**Pull request by Xiaro**
1. Refactored module base to use variable config
2. Refactored config system for cleaner code
3. Moved configs to kamiblue/config
4. Added server presets

Other:
Adapter for old config (before #1510) will be after this. And since we are now still in beta so I am not making adapter for the kamiblue/config directory change.

### CODE REVIEW

**Comment by l1ving** [comment]
If there isn't an adapter for waypoints / macros / friends I'll do it myself before merging. I'd also like to make a proper GUI for managing the configs, don't know if I'll do it this pull

**Comment by Xiaro** [comment]
> If there isn't an adapter for waypoints / macros / friends I'll do it myself before merging. I'd also like to make a proper GUI for managing the configs, don't know if I'll do it this pull

bruh there isn't any changes for them at all

**Comment by l1ving** [comment]
Well I haven't reviewed it yet so I didn't know that.

File: src/main/java/me/zeroeightsix/kami/module/modules/client/CommandConfig.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ object CommandConfig : Module(
         BackgroundScope.launchLooping("Config Auto Saving", 60000L) {
             if (autoSaving.value && mc.currentScreen !is KamiClickGui && timer.tick(savingInterval.value.toLong())) {
                 if (savingFeedBack.value) MessageSendHelper.sendChatMessage("Auto saving settings...")
-                ConfigUtils.saveConfig(ModuleConfig)
+                ConfigManager.save(ModuleConfig)
```
Comment (l1ving) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/CommandConfig.kt:44]: Shouldn't this save all configs?

File: src/main/java/me/zeroeightsix/kami/command/commands/ConfigCommand.kt [hunk 2]
```diff
@@ -1,77 +1,160 @@
...
+
+        return if (ip == null || mc.isIntegratedServerRunning) {
+            MessageSendHelper.sendWarningMessage("You are not in a server!")
+            null
+        } else {
+            ip
+        }
+    }
+
+    private fun confirm(name: String): Boolean {
```
Comment (l1ving) [inline src/main/java/me/zeroeightsix/kami/command/commands/ConfigCommand.kt:150]: You're only using the arg name without the literal itself, meaning if someone does
```
;config copy config1 // the user will be asked to confirm deletion
;config delete config1 // the user will not be asked to confirm deletion
```
the confirm name should include the literal arg name

**Comment by l1ving** [comment]
Tested, we can add adapter for pre-gui config later

**Pull request merged**